server/src/sas/LexerEx.ts

```diff
@@ -225,7 +231,7 @@ export class LexerEx {
         i++;
         block = blocks[i];
       } while (block && !this._isBefore(pos, this._startPos(block))); // |[]
-      return --i;
+      return block ? --i : -1;
```

Comment:
Hi, may I know the reason for this change? If there's not a `i+1` block, the `i` block is still valid.
There's a problem with current implementation. With code
```
data a;
  a;
run;
data b;
  b;
run;
```
If I put cursor at the start of the line `data b` (line 4 col 1), click `Run Region` from context menu, the `data b` is not run.

Reply:
@scnwwu  In the recursive senario, if a user clicks the firstline **"*region"**, and run region, it will search the block recursively. First it finds the *region block, then it will check every children blocks [data, proc]. If I let "i block" is still valid when there is not a i-1 block, it will then match the data block instead of the outer ***region** block. This is a fault.
```
*region;
data _null_;
run; 
proc print data=sashelp.cars;
run;
*endregion;
```

Reply:
Hi @jingjiajie,
There's another problem. With below code, put cursor at line 3 col 1, run region will run `*region` which is wrong.
```
*region;
proc print data=sashelp.class;
run;proc print data=sashelp.cars;
*endregion;
```

Reply:
Fixed